test/backend/routes.spec.js

```diff
@@ -198,60 +223,192 @@ describe('Routes - ', () => {
+             * irrespective of bad plotly-domain.
+             */
+            saveSetting('PLOTLY_API_DOMAIN', 'bad-domain.plot.ly');
+
+            GET('settings')
+            .then(res => {
+                assert.equal(res.status, 200);
+                done();
+            }).catch(done);
+        });
+    });
+
+    it('makes request to plotly if accessToken expired', function(done) {
+        saveSetting('ALLOWED_USERS', [username]);
+
+
+        // set access-token expiry of 1 sec:
+        accessTokenExpiryStub.returns(1);
+
+        POST('oauth2', {access_token: accessToken})
+        .then(res => {
+
+            // This ensures that any requests to plotly fail so that we can catch them
+            saveSetting('PLOTLY_API_DOMAIN', 'bad-domain.plot.ly');
+
+            setTimeout(() => {
+                GET('settings')
+                .then(res => res.json().then(json => {
+                    assert.equal(res.status, 500);
+                    assert.equal(json.error.message,
+                                 ('request to ' +
+                                 'https://bad-domain.plot.ly/v2/users/current failed, ' +
+                                 'reason: getaddrinfo ENOTFOUND ' +
+                                 'bad-domain.plot.ly bad-domain.plot.ly:443'));
+                    done();
+                })).catch(done);
+            }, 2000);
+        });
+    });
+
+    it('prevents user from accessing urls when revoked from ALLOWED_USERS', function(done) {
+        saveSetting('ALLOWED_USERS', [username]);
+
+
+        // set access-token expiry of 3 sec:
+        accessTokenExpiryStub.returns(3);
+
+        POST('oauth2', {access_token: accessToken})
+        .then(res => {
+            // request should be allowed:
+            GET('settings').then(res => {
+              console.log('status1:' + res.status);
+              assert.equal(res.status, 200);
+            });
+        })
+        .then(saveSetting('ALLOWED_USERS', []))
+        .then(() => {
+
+            // Any requests should fail after revoking user access
+            GET('settings').then(res => {
+                console.log('status2:' + res.status);
+                assert.equal(res.status, 403);
+                done();
+            });
+
+        }).catch(done);
+    });
+
+});
+
+describe('Routes - ', () => {
+    beforeEach((done) => {
+        servers = new Servers({createCerts: false, startHttps: false});
+        servers.httpServer.start();
+
+        // cleanup
+        ['CONNECTIONS_PATH', 'QUERIES_PATH', 'SETTINGS_PATH'].forEach(file => {
+            try {
+                fs.unlinkSync(getSetting(file));
+            } catch (e) {
+            }
+        });
+
+        // Save some connections to the user's disk
+        saveSetting('USERS', [{
+            username, apiKey
+        }]);
+
+        // Login the user:
+        saveSetting('ALLOWED_USERS', [username]);
+        setCookies(done);
```

Comment:
since this calls `done`, should this be at the very end of this `beforeEach` function? if this were synchronous, then the following lines wouldn't get called